### PR TITLE
Auto-delete .map extensions not in manifest

### PR DESCRIPTION
**Pull request by ghost**
I've made some small changes to add in a feature I found useful (and hope others will too).

I was using the plugin with the **gulp-sourcemaps** plugin and **gulp-rev**. When creating a manifest, the source maps are not included. A simple solution to also delete the source maps is to add the parameter **deleteMapExtensions** and lookup files that have an extra .map extension at the end to also delete those files. This could be more fully featured, such as specifying a directory for the source maps relative to the file being deleted, or reading the files themselves for comments on the source map location - but this was enough for my needs. I hope it helps someone else too.

### CODE REVIEW

**Comment by silverbackdan** [comment]
It turns out that I needed to separate the rev.manifest function out to a single process at the end of my tasks. Lots of processed were trying to write and merge the manifest at the same time. As a result my manifest will now have the .map files in it - I'm not sure how useful this feature would be but I'll leave the pull request here in case you do want it integrated.

File: index.js [hunk 1]
```diff
@@ -23,13 +23,37 @@ function revDel(options, cb) {
...
 		if (options.base) {
 			oldFiles = _.map(oldFiles, function (file) {
 				return path.join(options.dest || options.base, file);
 			});
 		}
-
+		if(options.deleteMapExtensions){
+			oldFiles.forEach(function(oldFile){
+				var mapPathCheck = oldFile+".map";
+				if (fs.existsSync(mapPathCheck)) {
```
Comment (callumacrae) [inline index.js]: Is there a way to do this without using `exists`?

Also, where possible, would you be able to use the coding style the rest of the code is using (single quotes, whitespace, etc).

Thanks for the PR!

**Comment by silverbackdan** [comment]
Sure! For some reason I thought existsSync wasn't being deprecated (just exists). I've updated that and I think changed the coding style to match. I'm new to pull requests and matching coding styles, please let me know if I've missed anything.

**Comment by callumacrae** [comment]
It was deprecated because checking whether something exists and then doing an action if it does is a bit dangerous - the file could be gone by the time the application gets on to the action.

The preferred approach is to just attempt the action, but with error handling. It looks like that would be difficult with this, though, because it's adding the file to an array. Sorry, should have checked the rest of the code properly before reviewing!

**Comment by silverbackdan** [comment]
Oh I see. I could take out the final check and just pass the variable for the map to be deleted into the oldFiles array. It might be worth leaving the first statSync check in place to avoid having to traverse the old manifest and find the unrevised filename. Alternatively I suppose I could just push both possible .map files that could possibly exist into the oldFiles array, but then we are traversing the old manifests array when we may not need to. Unless you think this feature request isn't really worth being added to your main repo - in which case I don't mind if you wanted to cancel it :) I've pushed changes to remove the second statSync (file exists) check if you want it though.

**Comment by callumacrae** [comment]
Sorry for the massive delay on this - merging now. If possible, would you be able to send another PR with a unit test? No worries if you don't want to.

**Pull request merged**